mod/logrotate/start.php

```diff
@@ -10,19 +10,11 @@
 function logrotate_init() {
 	$period = elgg_get_plugin_setting('period', 'logrotate');
 	$delete = elgg_get_plugin_setting('delete', 'logrotate');
-	switch ($period) {
-		case 'weekly':
-		case 'monthly' :
-		case 'yearly' :
-			break;
-		default:
-			$period = 'monthly';
-	}
 
 	// Register cron hook for archival of logs
 	elgg_register_plugin_hook_handler('cron', $period, 'logrotate_archive_cron');
```

Comment:
Don't query for settings. Register for all and match type against settig and terminate early.